notebooks/220-license-plate-recognition/220-license-plate-recognition.ipynb

```diff
@@ -0,0 +1,483 @@
+    "\n",
+    "#### Notes:\n",
+    "Since the license plate image could be any size, we need to use ImageLoader to resize it to fit the model input requirements.\n",
+    "Augmentation method on image is optional and it may lead to the wrong recognition result. Thus, we recommend you to use it under special conditions like image overexposure or too dark."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def ShowImage(\n",
+    "    OriginImage: np.ndarray, InputImage: np.ndarray\n",
+    ") -> matplotlib.figure.Figure:\n",
+    "    \"\"\"\n",
+    "    Visualise how the image we processed.\n",
+    "\n",
+    "    Args:\n",
+    "        OriginImage (np.ndarray): Any size of color image.\n",
+    "        InputImage (np.ndarray): The input of the model before dimension transposed and expanded.\n",
+    "\n",
+    "    Returns:\n",
+    "        Matplotlib figure.\n",
+    "    \"\"\"\n",
+    "    figure, axis = plt.subplots(1, 2, figsize=(18, 9), squeeze=False)\n",
+    "    \n",
+    "    # adjust the image channels to the correct order.\n",
+    "    b,g,r = cv2.split(OriginImage)\n",
+    "    OriginImage = cv2.merge((r,g,b))\n",
+    "    axis[0, 0].imshow(OriginImage)\n",
+    "    axis[0, 0].set_title(\"OriginImage\")\n",
+    "    \n",
+    "    # adjust the image channels to the correct order.\n",
+    "    b,g,r = cv2.split(InputImage)\n",
+    "    InputImage = cv2.merge((r,g,b))\n",
+    "    axis[0, 1].imshow(InputImage)\n",
+    "    axis[0, 1].set_title(\"InputImage\")\n",
+    "    \n",
+    "    return figure\n",
+    "\n",
+    "\n",
+    "def ImageLoader(ImagePath: str, Augmentation: str=\"None\") -> np.ndarray:\n",
+    "    \"\"\"\n",
+    "    Process the image if neccessary and then\n",
+    "    resize its shape to fit the input requirements.\n",
+    "\n",
+    "    Args:\n",
+    "        ImagePath (string): Relative storage path of color image.\n",
+    "        Augmentation (string): Image augmentation method, if 'None', do nothing.\n",
+    "                    Including 'None', 'Laplace', 'Log', 'Gamma' and 'EquaHist'.\n",
+    "\n",
+    "    Returns:\n",
+    "        input_img (np.ndarray): The input of the model. The dimension is 1x3x24x94.\n",
+    "\n",
+    "    \"\"\"\n",
+    "    ori_img = cv2.imread(ImagePath,1)\n",
+    "\n",
+    "    # if gray image passed in, the programme will stop.\n",
+    "    assert len(ori_img.shape) == 3, \"Failed to load image.\"\n",
+    "    print(f\"Origin image shape: {ori_img.shape}\")\n",
+    "\n",
+    "    # if invalid augmentation method passed in, the programme will stop.\n",
+    "    assert Augmentation in [\n",
+    "        \"None\",\n",
+    "        \"Laplace\",\n",
+    "        \"Log\",\n",
+    "        \"Gamma\",\n",
+    "        \"EquaHist\",\n",
+    "    ], \"Invalid Augmentation.\"\n",
+    "\n",
```

Comment:
Notebooks are meant to be as simple as possible. Augmentation is not key feature for this notebook, can you reduce number of options for augmentation to keep it only as practical example?